### DISCUSSION TITLE
Single Code Base Monorepo- possible?

### DISCUSSION THREAD

**Discussion by EinfachHans** [ionic-team/capacitor#4335] [Q&A] [answered]
Hey there 👋🏼

first: i'm not a Capictor User yet. The Reason ist that the apps that i'm building currently all use Google Maps and because i only wan't to use the native SDK's (not JS) and the [Google Maps Plugin](https://github.com/mapsplugin/cordova-plugin-googlemaps/issues/2781) is not finished yet, i can't. 

I have done some small Capacitor Projects as Test but i'm not very into yet.

**Now my Question**: I'm currently developing a new App-System, where i have one App-Code, which i can deploy to unlimited Customer Apps (planned (hopefully) some hundert). Is this even possible with Capacitor? With Cordova i can use Hooks to replace related Stuff (`config.xml` values like id, name, etc., Icon Splash Screen) before build and then run the cordova Build as Cordova generates the native App Folders by zero again.

As far as i understand Capacitor doesn't do this like this and isn't able to create the full native projects from zero with all related native changes (plist values for example).

I'm i correct? So when doing this is Cordova the only Possibility? 🤔

Also i'm planning to use Appflow to build the Apps as it offers Monorepo Support (i don't know if this is relevant)

**Comment by eric-horodyski** [comment, marked as answer]
I've certainly built monorepos with Capacitor -- so it is possible:

* https://github.com/eric-horodyski/nx-ng-monorepo
* https://github.com/eric-horodyski/lerna-cap-react-appflow

The real question - in my opinion - is whether or not you want your white-label solution to have each client project persisted to source control or not. I haven't done it myself, but I'm sure you could script a process not dissimilar to what you have with your Cordova hooks; wipe out the existing Android/iOS project folders, pick a new client folder to build with, script Capacitor to add the platforms and build. 

But you wouldn't be gaining a more efficient process, if that's what you're looking for. Personally I'd rather keep all client projects separated because if experience has taught me anything it would be that _someone_ will sign off on a one-off customization for Client X and it'll be more complicated to conditionally suppress it than it would to manage multiple projects, especially in a monorepo.

**Reply by EinfachHans** [reply]
Thanks for your Thoughts! 😊 In our case white labeling is the only possible solution. All apps will (and should) act and look the same - with some small adjustments and a own database

i currently ended up creating a script which modifys all native Files (gradle, manifest, plist, etc) to adjust stuff like bundle id, version etc.

I will then configure appflow to use many apps and override the build command to prepare the specific app before the build
